src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java

```diff
@@ -138,6 +139,24 @@ class StoreAppender extends AbstractCloseable
         queue.addCloseListener(this);
     }
 
+    void bindToCreatingThread() {
+        this.boundToCreatingThread = true;
+    }
+
+    @Override
+    public void singleThreadedCheckReset() {
+        if (boundToCreatingThread)
+            throw new UnsupportedOperationException("This appender cannot be used by any thread but the creating thread");
+        super.singleThreadedCheckReset();
+    }
+
+    @Override
+    public void singleThreadedCheckDisabled(boolean singleThreadedCheckDisabled) {
+        if (boundToCreatingThread)
+            throw new UnsupportedOperationException("This appender cannot be used by any thread but the creating thread");
+        super.singleThreadedCheckDisabled(singleThreadedCheckDisabled);
+    }
```

Comment:
This allows thread-local Appenders to prevent bypassing of the single thread protections.